app/controllers/application_controller.rb

```diff
@@ -79,6 +96,9 @@ def flow_update_selection
   def flow_sync_order
     return unless @order && @order.id
 
+    # maybe delete order in flow if we have sync id
+    return if @order.line_items.length == 0
```

Comment:
Note: we do delete the order in shopify once the last line is removed. Fine either way